File: android/activitysources/src/main/java/com/fjuul/sdk/activitysources/entities/internal/healthconnect/HCCaloriesDataPoint.java
package com.fjuul.sdk.activitysources.entities.internal.healthconnect;

import java.util.Date;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

public class HCCaloriesDataPoint extends HCDataPoint {
    private final float totalCalories;
    private final float activeCalories;

    public HCCaloriesDataPoint(@NonNull Date start, @NonNull Date end, float totalCalories, float activeCalories, @Nullable String dataSource) {
        super(start, end, dataSource);
        this.totalCalories = totalCalories;
        this.activeCalories = activeCalories;
    }

    public float getTotalCalories() {
        return totalCalories;
    }

    public float getActiveCalories() {
        return activeCalories;
    }
} 

File: android/activitysources/src/main/java/com/fjuul/sdk/activitysources/entities/internal/healthconnect/HCCaloriesDataPoint.kt
package com.fjuul.sdk.activitysources.entities.internal.healthconnect

import java.time.Instant

/**
 * Represents a total calories burned record from Health Connect.
 */
data class HCTotalCaloriesDataPoint(
    override val start: Instant,
    override val value: Double,
    override val dataSources: Set<String>
) : HCDailyDataPoint()

/**
 * Represents an active calories burned record from Health Connect.
 */
data class HCActiveCaloriesDataPoint(
    override val start: Instant,
    override val value: Double,
    override val dataSources: Set<String>
) : HCDailyDataPoint()
